lib/service/service.go

```diff
@@ -967,8 +967,12 @@ func NewTeleport(cfg *servicecfg.Config) (*TeleportProcess, error) {
 	imClient := cfg.InstanceMetadataClient
 	if imClient == nil {
 		imClient, err = cloud.DiscoverInstanceMetadata(supervisor.ExitContext())
-		if err != nil && !trace.IsNotFound(err) {
-			return nil, trace.Wrap(err)
+		if err == nil {
+			cfg.Logger.InfoContext(supervisor.ExitContext(),
+				"Found an instance metadata service. Teleport will import labels from this cloud instance.",
+				"type", imClient.GetType())
+		} else if !trace.IsNotFound(err) {
+			cfg.Logger.ErrorContext(supervisor.ExitContext(), "Error looking for cloud instance metadata.", "error", err)
```

Comment:
As a question to the above, why is it OK to keep going in this case (and the ones below)?

Do we have test coverage for this?

Reply:
Sometimes a user doesn't need labels to be imported, so they don't configure their instance to do so. An error here isn't a problem for them and shouldn't prevent Teleport from starting. (all of this is now in a comment)

Reply:
Thanks, that helps.